From 9dd52f80b94bb644797b6775dd81fd02e57117be Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Mon, 4 Jun 2012 20:51:57 +0800
Subject: ENGR00211169-2 [SMP]Need to update loops_per_jiffy when cpufreq is
 changed

Need to update loops_per_jiffy when cpufreq is changed, our platform
only change the loops_per_jiffy of per_cpu cpu_data, we need to update
it to global loops_per_jiffy which is used for udelay, this is for pure
cpu freq change.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/plat-mxc/cpufreq.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/plat-mxc/cpufreq.c b/arch/arm/plat-mxc/cpufreq.c
index 3a0b818..cd13e4b 100755
--- a/arch/arm/plat-mxc/cpufreq.c
+++ b/arch/arm/plat-mxc/cpufreq.c
@@ -361,13 +361,15 @@ static int mxc_set_target(struct cpufreq_policy *policy,
 		per_cpu(cpu_data, i).loops_per_jiffy =
 		cpufreq_scale(per_cpu(cpu_data, i).loops_per_jiffy,
 					freqs.old, freqs.new);
+	/* Update global loops_per_jiffy to cpu0's loops_per_jiffy,
+	 * as all CPUs are running at same freq */
+	loops_per_jiffy = per_cpu(cpu_data, 0).loops_per_jiffy;
 #endif
 	for (i = 0; i < num_cpus; i++) {
 		freqs.cpu = i;
 		cpufreq_notify_transition(&freqs, CPUFREQ_POSTCHANGE);
 	}
 
-
 	return ret;
 }
 
-- 
1.7.9.5

